llvm/lib/Target/AMDGPU/DSInstructions.td

```diff
@@ -969,7 +969,7 @@ multiclass DSReadPat_t16<DS_Pseudo inst, ValueType vt, string frag> {
   }
 
   let OtherPredicates = [NotLDSRequiresM0Init] in {
-    foreach p = [NotHasTrue16BitInsts, UseFakeTrue16Insts] in
+    foreach p = [NotHasTrue16BitInsts, UseFakeTrue16Insts, UseLoadStoreTrue16WithSramECC] in
```

Comment:
I really don't understand why you would have multiple copies of the same pattern with different preditcates. Surely it is better to have a single pattern with a single predicate equivalent to "NotHasTrue16BitInsts **OR** UseFakeTrue16Insts **OR** UseLoadStoreTrue16WithSramECC"?

(Pre-existing problem, but this patch makes it worse.)

Reply:
Also, is this really required for DS instructions? I thought it applied to all other VMEM instructions but **not** DS.

Reply:
It will need a separate predicate on top of your change though, which will only apply to memory instructions.